# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/noad-backtrace.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- noad-0.5.2/main.cpp.orig    2004-12-17 22:26:42.000000000 +0100
+++ noad-0.5.2/main.cpp 2005-02-16 21:57:35.875947808 +0100
@@ -24,7 +24,6 @@
 #include <time.h>
 #include <sys/wait.h>
 #include <signal.h>
-#include <execinfo.h>
 #include "noad.h"
 #include "tnoad.h"

--- noad-0.5.2/tools.cpp.orig	2005-02-16 21:27:30.000000000 +0100
+++ noad-0.5.2/tools.cpp	2005-02-16 21:30:33.661561880 +0100
@@ -16,7 +16,6 @@
 #include <sys/vfs.h>
 #include <time.h>
 #include <unistd.h>
-#include <execinfo.h>
 
 extern int SysLogLevel;
 
@@ -1042,6 +1041,11 @@
 
 void show_stackframe(bool bFork) 
 {
+  char *msg = "[bt] Execution path: <Sorry, stackframe backtrace not available with uClibc>";
+  if (!bFork)
+    fprintf(stderr, "%s\n", msg);
+  syslog(LOG_INFO, msg);
+/*
   void *trace[32];
   char **messages = (char **)NULL;
   int i, trace_size = 0;
@@ -1057,5 +1061,6 @@
       fprintf(stderr, "[bt] %s\n", messages[i]);
     syslog(LOG_INFO, "[bt] %s\n", messages[i]);
   }
+*/
 }
 
